pandas/core/internals/blocks.py

```diff
@@ -166,6 +169,16 @@ def __init__(self, values, placement, ndim: int):
         self.values = self._maybe_coerce_values(values)
 
         if self._validate_ndim and self.ndim and len(self.mgr_locs) != len(self.values):
+            if (
+                is_datetime64tz_dtype(values.dtype)
+                and self.ndim == 2
+                and values.ndim == 1
+                and len(self.mgr_locs) == 1
+            ):
+                # Passed by fastparquet; TODO: warn?
+                self.values = self.values.reshape(1, -1)
```

Comment:
@TomAugspurger fastparquet's usage of internals is causing problems here.  (this workaround to fix a fastparquet test breaks other tests).  Thoughts on how to address this?

Reply:
Maybe check with @martindurant. It'd be good to provide some context (like what lines in fastparquet are causing issues and why)

Reply:
Could you keep DatetimeTZBlock store a 1D array?

Reply:
I tentatively think I have a patch for this particular problem, but this is bound to keep coming up.

> It'd be good to provide some context (like what lines in fastparquet are causing issues and why)

```
/usr/share/miniconda/envs/pandas-dev/lib/python3.7/site-packages/fastparquet/dataframe.py:179: in empty
    type(block.values)(values, dtype=block.values.dtype)
pandas/core/internals/blocks.py:329: in make_block_same_class
    return type(self)(values, placement=placement, ndim=self.ndim)
pandas/core/internals/blocks.py:167: in __init__
    self.ndim = self._check_ndim(values, ndim)
```

This branch changes what `DatetimeTZBlock.__init__` expects, so the call to make_block_same_class raises.

> Could you keep DatetimeTZBlock store a 1D array?

that would defeat a big chunk of the purpose

Reply:
Sorry, what am I being asked? fastparquet is of course happy to change code if required, but that would still cause breakages for someone who updates pandas and not fastparquet. If pandas had a public way to create an empty dataframe and constituent numpy arrays of a given set of data types, we would happily use it!

Reply:
> Sorry, what am I being asked?

ATM nothing, apologies for the premature ping.  We need to sort out exactly what pseudo-public API to ask downstream libraries to use